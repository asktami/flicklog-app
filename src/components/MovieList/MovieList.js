import React from 'react';

import MovieListItem from '../../components/MovieListItem/MovieListItem';

const MovieList = (props) => {
	const { movies, onSavedList = false, showWatchListButton = '' } = props;

	const renderMovieList = () => {
		if (movies && movies.length > 0) {
			let unique_movie_list = [];
			return movies.map((movie, i) => {
				// can have more than 1 review record per user per movie
				// only for watchList and reviewList only show 1 instance of movie
				// for movie search lists show whatever is found

				if (
					onSavedList === true &&
					!unique_movie_list.includes(movie.movie_id)
				) {
					unique_movie_list.push(movie.movie_id);
					return (
						<MovieListItem
							key={movie.id}
							movie={movie}
							showWatchListButton={showWatchListButton}
						/>
					);
				} else if (onSavedList !== true) {
					return (
						<MovieListItem
							key={movie.id}
							movie={movie}
							showWatchListButton={showWatchListButton}
						/>
					);
				} else {
					return null;
				}
			});
		} else {
			return 'No results.';
		}
	};

	return renderMovieList();
};
export default MovieList;
