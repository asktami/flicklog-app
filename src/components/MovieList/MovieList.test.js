import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


import MovieList from './MovieList';

describe(`MovieList component`, () => {
	const props = {
		match: {
			params: {
				movies: [
						   {
						   poster_path: '/vfrQk5IPloGg1v9Rzbh2Eg3VGyM.jpg',
						   backdrop_path: '/AmR3JG1VQVxU8TfAvljUhfSFUOx.jpg',
						   title: 'Alien',
						   release_date: '1979-05-25',
						   vote_average: 8.1,
						   vote_count: 5694,
						   id: 348
						   }
						]
			}
		}
	};
	

	it('renders a MovieList by default', () => {
		const wrapper = shallow(<MovieList {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders the MovieList given props', () => {
		const wrapper = shallow(<MovieList {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
