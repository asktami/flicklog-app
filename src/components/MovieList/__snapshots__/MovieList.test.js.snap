// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList component renders a MovieList by default 1`] = `"No results."`;

exports[`MovieList component renders the MovieList given props 1`] = `"No results."`;
